src/shared/parameter_v2/scripts/config_yaml_loader.py

```diff
@@ -0,0 +1,288 @@
+            included configs and a list of defined parameters.
+        {
+            "ai.yaml": {
+                "includes": ["passing.yaml", "plays.yaml"],
+                "parameters" :
+                [{
+                    "int": {
+                        "name": "example_int", "value": 10, "min": 0, "max": 20
+                    },
+                    ...
+                }]
+            },
+
+            "plays.yaml": {
+                "includes": ["defense.yaml", "offense.yaml"],
+                ...
+            }
+            ...
+        }
+
+        :rtype: dict
+
+        """
+        # load yaml into config_metadata
+        config_metadata = ConfigYamlLoader.__load_yaml_into_dict(yaml_paths)
+
+        # validate schema
+        ConfigYamlLoader.__validate_config_metadata(config_metadata)
+
+        # detect cycles
+        ConfigYamlLoader.__detect_cycles_in_config_metadata(config_metadata)
+
+        # valid config ready to be generated
+        return config_metadata
+
+    @staticmethod
+    def __load_yaml_into_dict(yaml_paths: list):
+        """Loads the yamls into an dictionary. Any errors while in the yaml
+        syntax will raise to the main thread. We also adjust how the dictionary
+        is stored for easier access later.
+
+        :raises: ConfigYamlMalformed
+        :param yaml_paths: the path to all the config yamls
+        :type yaml_paths: list of str
+        :returns: config_medata dict representing the data to generate
+        :rtype: dict
+
+        """
+        raw_config_metadata = {}
+
+        for filename in yaml_paths:
+            with open(filename, "r") as param_yaml:
+
+                try:
+                    # extract config name from filename
+                    _, tail = os.path.split(filename)
+
+                    # safe load yaml into dictionary
+                    raw_config_metadata[tail] = list(yaml.safe_load_all(param_yaml))
+
+                    if len(raw_config_metadata[tail]) == 1:
+
+                        # include only in file
+                        if isinstance(raw_config_metadata[tail][0], dict):
+                            raw_config_metadata[tail] = {
+                                "include": raw_config_metadata[tail][0]["include"]
+                            }
+
+                        # parameter definitions only in file
+                        if isinstance(raw_config_metadata[tail][0], list):
+                            raw_config_metadata[tail] = {
+                                "parameters": raw_config_metadata[tail][0]
+                            }
+
+                    elif len(raw_config_metadata[tail]) == 2:
+
+                        # include and param definition in file
+                        raw_config_metadata[tail] = {
+                            "include": raw_config_metadata[tail][0]["include"],
+                            "parameters": raw_config_metadata[tail][1],
+                        }
```

Comment:
Should there be an `else` case just in case we do something weird like add another list or dict at the top-level?

Reply:
good point, added